examples/code_repository/README.md

```diff
@@ -0,0 +1,93 @@
+# Using a code repository
+
+Code repositories enable ZenML to keep track of the code version that you use for your
+pipeline runs. Additionally, running a pipeline which is tracked in a registered code
+repository can speed up the Docker image building for containerized stack components.
+
```

Comment:
This really needs a more visual explanation with a diagram. i.e. why would people use this?

Reply:
also probably needs an extra few sentences just explaining / reminding people why it might be useful to be tracking their code & pairing it with the rest of their pipeline runs etc in the context of MLOps.